test/ibmq/test_ibmq_job.py

```diff
@@ -296,13 +297,55 @@ def test_retrieve_job_error(self, provider):
     def test_retrieve_jobs_status(self, provider):
         """Test retrieving jobs filtered by status."""
         backend = provider.get_backend('ibmq_qasm_simulator')
-        job_list = provider.backends.jobs(backend_name=backend.name(),
-                                          limit=5, skip=0, status=JobStatus.DONE)
-
-        self.assertTrue(job_list)
-        for job in job_list:
+        # Get the most recent jobs that are done.
+        backend_jobs = backend.jobs(limit=10, skip=0, status=JobStatus.DONE)
+        backend_jobs_status_filtered = backend.jobs(limit=10, skip=0, status=[JobStatus.DONE])
+        backend_jobs_str_filtered = backend.jobs(limit=10, skip=0, status=['DONE'])
+        self.assertTrue(backend_jobs)
+        self.assertTrue(backend_jobs_status_filtered)
+        self.assertTrue(backend_jobs_str_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_status_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_str_filtered)
+
+        for job in backend_jobs:
             self.assertTrue(job.status() is JobStatus.DONE)
 
+    @requires_provider
+    def test_retrieve_all_job_statuses(self, provider):
+        """Test retrieving jobs filtered by all possible job statuses."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        # pylint: disable=unnecessary-comprehension
+        all_job_statuses = [status for status in JobStatus]
+        # Get the most recent jobs with any job status.
+        backend_jobs_filtered = backend.jobs(limit=10, status=all_job_statuses)
+        backend_jobs = backend.jobs(limit=10)
+        self.assertTrue(backend_jobs)
+        self.assertTrue(backend_jobs_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_filtered)
```

Comment:
This assert will fail if another test instance submitted another job in between the two `jobs()` calls.